tests/components/iskra/test_config_flow.py

```diff
@@ -0,0 +1,233 @@
+
+    # Test successful authentication
+    mock_pyiskra_rest.side_effect = None
+
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        user_input={CONF_USERNAME: USERNAME, CONF_PASSWORD: PASSWORD},
+    )
+
+    # Test successful Rest API configuration
+    assert result["type"] is FlowResultType.CREATE_ENTRY
+    assert result["title"] == f"{SG_MODEL} {SERIAL}"
+    assert result["data"][CONF_HOST] == HOST
+    assert result["data"][CONF_PROTOCOL] == "rest_api"
+    assert result["data"][CONF_USERNAME] == USERNAME
+    assert result["data"][CONF_PASSWORD] == PASSWORD
+
+
+async def test_user_modbus(hass: HomeAssistant, mock_pyiskra_modbus) -> None:
+    """Test the user flow with Modbus TCP protocol."""
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"source": SOURCE_USER},
+    )
+
+    # Test if user form is provided
+    assert result["type"] is FlowResultType.FORM
+    assert result["step_id"] == "user"
+
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        user_input={CONF_HOST: HOST, CONF_PROTOCOL: "modbus_tcp"},
+    )
+
+    # Test if propmpted to enter port and address
+    assert result["type"] is FlowResultType.FORM
+    assert result["step_id"] == "modbus_tcp"
+
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        user_input={
+            CONF_PORT: MODBUS_PORT,
+            CONF_ADDRESS: MODBUS_ADDRESS,
+        },
+    )
+
+    # Test successful Modbus TCP configuration
+    assert result["type"] is FlowResultType.CREATE_ENTRY
+    assert result["title"] == f"{PQ_MODEL} {SERIAL}"
+    assert result["data"][CONF_HOST] == HOST
+    assert result["data"][CONF_PROTOCOL] == "modbus_tcp"
+    assert result["data"][CONF_PORT] == MODBUS_PORT
+    assert result["data"][CONF_ADDRESS] == MODBUS_ADDRESS
+
+
+@pytest.mark.parametrize(
+    ("protocol"),
+    [
+        ("rest_api"),
+        ("modbus_tcp"),
+    ],
+)
+async def test_abort_if_already_setup(
+    hass: HomeAssistant, mock_pyiskra_rest, mock_pyiskra_modbus, protocol
+) -> None:
+    """Test we abort if Iskra is already setup."""
+
+    MockConfigEntry(domain=DOMAIN, unique_id=SERIAL).add_to_hass(hass)
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"source": SOURCE_USER},
+        data={CONF_HOST: HOST, CONF_PROTOCOL: protocol},
+    )
+
+    if protocol == "modbus_tcp":
+        assert result["type"] is FlowResultType.FORM
+        assert result["step_id"] == "modbus_tcp"
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            user_input={
+                CONF_PORT: MODBUS_PORT,
+                CONF_ADDRESS: MODBUS_ADDRESS,
+            },
+        )
+
+    assert result["type"] is FlowResultType.ABORT
+    assert result["reason"] == "already_configured"
```

Comment:
Not sure if I told this test to be parametrized, but we should not have if statements in the code

Reply:
Maybe I forgot that it required an extra step, if that's the case, sorry

Reply:
np, changed it back same for error test